doc-content/drools-docs/src/main/asciidoc/DecisionEngine/phreak-sequential-mode-con.adoc

```diff
@@ -0,0 +1,83 @@
+
+Sequential mode applies to only stateless KIE sessions because stateful KIE sessions inherently use data from previously invoked KIE sessions. If you use a stateless KIE session and you want the execution of rules to influence subsequent rules in the agenda, then do not enable sequential mode. Sequential mode is disabled by default in the {DECISION_ENGINE}.
+
+To enable sequential mode, use one of the following options:
+
+* Set the system property `drools.sequential` to `true`.
+* Enable sequential mode while creating the KIE base in the client code:
++
+[source,java]
+----
+KieServices services = KieServices.Factory.get();
+KieContainer container = services.newKieContainer(releaseId);
+
+KieBaseConfiguration conf = KieServices.Factory.get().newKieBaseConfiguration();
+conf.setOption(SequentialOption.YES);
+
+KieBase kieBase = kc.newKieBase(conf);
+----
+
+To configure sequential mode to use a dynamic agenda, use one of the following options:
+
+* Set the system property `drools.sequential.agenda` to `dynamic`.
+* Set the sequential agenda option while creating the KIE base in the client code:
++
+[source,java]
+----
+KieServices services = KieServices.Factory.get();
+KieContainer container = services.newKieContainer(releaseId);
+
+KieBaseConfiguration conf = KieServices.Factory.get().newKieBaseConfiguration();
+conf.setOption(SequentialAgendaOption.DYNAMIC);
+
+KieBase kieBase = kc.newKieBase(conf);
+----
+
+When you enable sequential mode, the {DECISION_ENGINE} evaluates rules in the following way:
+
+. Rules are ordered by salience and position in the rule set.
+. An element for each possible rule match is created. The element position indicates the execution order.
+. Node memory is disabled, with the exception of the right-input object memory.
+. The left-input adapter node propagation is disconnected and the object with the node is referenced in a `Command` object. The `Command` object is added to a list in the working memory for later execution.
+. All objects are asserted, and then the list of `Command` objects is checked and executed.
+. All matches that result from executing the list are added to elements based on the sequence number of the rule.
+. The elements that contain matches are executed in a sequence. If you set a maximum number of rule executions, the {DECISION_ENGINE} activates no more than that number of rules in the agenda for execution.
+
+In sequential mode, the `LeftInputAdapterNode` node creates a `Command` object and adds it to a list in the working memory of the {DECISION_ENGINE}. This `Command` object contains references to the `LeftInputAdapterNode` node and the propagated object. These references stop any left-input propagations at insertion time so that the right-input propagation never needs to attempt to join the left inputs. The references also avoid the need for the left-input memory.
```

Comment:
Agreed, but the purpose of this doc is to serve as a decision engine reference for more info. It's not necessarily meant to be cohesive, and some parts are more technical than others.. Yes, I'm in favor of a clearer re-write of this section, if an SME is up to the task :) It will need to be done by an SME.

Reply:
Agree, was just pointing it out so more people are aware that it is a bit untidy :)

Reply:
Ack.